src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutItem.cs

```diff
@@ -73,5 +139,685 @@ internal void InvokeClick()
+					CommandingHelpers.ClearBindingIfSet(oldCommand, this, AutomationProperties.HelpTextProperty);
+					CommandingHelpers.ClearBindingIfSet(oldCommand, this, ToolTipService.ToolTipProperty);
+				}
+			}
+
+			// Subscribe to the CanExecuteChanged event on the new value
+			if (pNewValue != null)
+			{
+				var spNewCommand = pNewValue as ICommand;
+
+				void OnCanExecuteUpdated(object sender, object args)
+				{
+					UpdateCanExecute();
+				}
+
+				spNewCommand.CanExecuteChanged += OnCanExecuteUpdated;
+				m_epCanExecuteChangedHandler = Disposable.Create(() => spNewCommand.CanExecuteChanged -= OnCanExecuteUpdated);
+
+				var newCommandAsUICommand = spNewCommand as XamlUICommand;
+
+				if (newCommandAsUICommand != null)
+				{
+					CommandingHelpers.BindToLabelPropertyIfUnset(newCommandAsUICommand, this, TextProperty);
+					CommandingHelpers.BindToIconPropertyIfUnset(newCommandAsUICommand, this, IconProperty);
+					CommandingHelpers.BindToKeyboardAcceleratorsIfUnset(newCommandAsUICommand, this);
+					CommandingHelpers.BindToAccessKeyIfUnset(newCommandAsUICommand, this);
+					CommandingHelpers.BindToDescriptionPropertiesIfUnset(newCommandAsUICommand, this);
+				}
+			}
+
+			// Coerce the button enabled state with the CanExecute state of the command.
+			UpdateCanExecute();
+		}
+
+		// Coerces the MenuFlyoutItem's enabled state with the CanExecute state of the Command.
+		 void UpdateCanExecute()
+		{
+			ICommand spCommand;
+			object spCommandParameter;
+			bool canExecute = true;
+
+			spCommand = Command;
+			if (spCommand != null)
+			{
+				spCommandParameter = CommandParameter;
+				canExecute = spCommand.CanExecute(spCommandParameter);
+			}
+
+			SuppressIsEnabled(!canExecute);
+		}
+
+		// Change to the correct visual state for the 
+		private protected override void ChangeVisualState(
+			// true to use transitions when updating the visual state, false
+			// to snap directly to the new visual state.
+			 bool bUseTransitions)
+		{
+			bool hasToggleMenuItem = false;
+			bool hasIconMenuItem = false;
+			bool hasMenuItemWithKeyboardAcceleratorText = false;
+			bool isKeyboardPresent = false;
+
+			MenuFlyoutPresenter spPresenter = GetParentMenuFlyoutPresenter();
+			if (spPresenter != null)
+			{
+				hasToggleMenuItem = spPresenter.GetContainsToggleItems();
+				hasIconMenuItem = spPresenter.GetContainsIconItems();
+				hasMenuItemWithKeyboardAcceleratorText = spPresenter.GetContainsItemsWithKeyboardAcceleratorText();
+			}
+
+			var bIsEnabled = IsEnabled;
+			var focusState = FocusState;
+			var shouldBeNarrow = GetShouldBeNarrow();
+
+			// We only care about finding if we have a keyboard if we also have a menu item with accelerator text,
+			// since if we don't have any menu items with accelerator text, we won't be showing any accelerator text anyway.
+			if (hasMenuItemWithKeyboardAcceleratorText)
+			{
+				// UNO TODO
+				// isKeyboardPresent = DXamlCore.GetCurrent().GetIsKeyboardPresent();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)